src/main/java/internet/store/dao/jdbc/OrderDaoJdbcImpl.java

```diff
@@ -0,0 +1,148 @@
+    public Order create(Order order) {
+        String query = "INSERT INTO orders (user_id) values (?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            statement.setLong(1, order.getUserId());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                order.setOrderId(resultSet.getLong(1));
+            }
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Order " + order + " was not created", ex);
+        }
+        return addProductsToOrder(order);
+    }
+
+    @Override
+    public Order update(Order order) {
+        deleteProductsfromOrder(order);
+        return addProductsToOrder(order);
+    }
+
+    @Override
+    public boolean delete(Long orderId) {
+        String query = "UPDATE orders SET deleted = true WHERE order_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, orderId);
+            return statement.executeUpdate() == 1;
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Order # " + orderId + " was not deleted", ex);
+        }
+    }
+
+    @Override
+    public Optional<Order> get(Long orderId) {
+        String query = "SELECT order_id, user_id, product_id, product_name, price FROM orders"
+                + " JOIN orders_products USING(order_id)"
+                + " JOIN products ON (product_id = products.id)"
+                + " WHERE order_id = ? AND orders.deleted = false;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, orderId);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                Long prodId = resultSet.getLong("product_id");
+                Long userId = resultSet.getLong("user_id");
+                Order order = new Order(userId);
+                order.setOrderId(orderId);
+                order.setProducts(retrieveProductData(resultSet));
+                return Optional.of(order);
+            }
+        } catch (SQLException ex) {
+            throw new DataProcessingException("No order # " + orderId + " found", ex);
+        }
+        return Optional.empty();
+    }
+
+    private boolean deleteProductsfromOrder(Order order) {
+        String query = "DELETE FROM orders_products WHERE order_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            return statement.executeUpdate() == 1;
```

Comment:
will executeUpdate() return 1 if you delete more than 1 row ?

Reply:
As did is based on my current logic (of shopping cart set in web-application) - with every push of button we delete one record. We might do deletion in batch, but this will be done either not in the web app or we will need to have kind of check-boxes there and then  - "delete" button. Don't have this functionality now in my application but should be ready for such changes, so agree with your comment.